jbpm-container-test/jbpm-container-test-suite/src/test/java/org/jbpm/test/container/test/ejbservices/tx/ETransactionTest.java

```diff
@@ -0,0 +1,397 @@
+        // processService.signalProcessInstance(processInstanceId, "start",
+        // "rfg");
+        // Thread.sleep(1000);
+        //
+        // ut.commit();
+        //
+        // Assertions.assertThat(hasNodeLeft(processInstanceId,
+        // "rfg")).isTrue();
+        // processService.signalProcessInstance(processInstanceId, "finish",
+        // null);
+        // processService.fireAllRules();
+        // Thread.sleep(1000);
+        // Assertions.assertThat(agenda.isRuleFired("dummyRule")).isTrue();
+        // Assertions.assertThat(hasProcessInstanceCompleted(processInstanceId)).isTrue();
+    }
+
+    @Test
+    public void testTimer() throws Exception {
+        Long processInstanceId = startProcessInstance(PROCESS_ID);
+
+        UserTransaction ut = InitialContext.doLookup(USER_TRANSACTION_NAME);
+        ut.begin();
+
+        try {
+            processService.signalProcessInstance(processInstanceId, "start", "timer");
+            Assertions.assertThat(hasNodeLeft(processInstanceId, "timer")).isTrue();
+        } catch (Exception e) {
+            ut.rollback();
+            throw e;
+        }
+
+        ut.rollback();
+
+        ut = InitialContext.doLookup(USER_TRANSACTION_NAME);
+        ut.begin();
+
+        try {
+            processService.signalProcessInstance(processInstanceId, "start", "timer");
+        } catch (Exception e) {
+            ut.rollback();
+            throw e;
+        }
+
+        ut.commit();
+
+        Thread.sleep(2000);             //to make sure the timer is completed
+
+        Assertions.assertThat(hasNodeLeft(processInstanceId, "timer")).isTrue();
+        Assertions.assertThat(hasNodeLeft(processInstanceId, "Timer")).isTrue();
+        processService.signalProcessInstance(processInstanceId, "finish", null);
+        Assertions.assertThat(hasProcessInstanceCompleted(processInstanceId)).isTrue();
+    }
+
+    // BZ-1150970
+    @BZ("1150970")
+    @Test
+    public void testUserTask() throws Exception {
+        Long processInstanceId = startProcessInstance(PROCESS_ID);
+
+        UserTransaction ut = InitialContext.doLookup(USER_TRANSACTION_NAME);
+        ut.begin();
+
+        List<TaskSummary> taskSummaries = null;
+        try {
+            processService.signalProcessInstance(processInstanceId, "start", "usertask");
+
+            taskSummaries = runtimeDataService.getTasksAssignedAsPotentialOwner("john", new QueryFilter());
+            Assertions.assertThat(taskSummaries).isNotNull().hasSize(1);
+            userTaskService.start(taskSummaries.get(0).getId(), "john");
+            userTaskService.complete(taskSummaries.get(0).getId(), "john", new HashMap<String, Object>());
+            Assertions.assertThat(hasNodeLeft(processInstanceId, "User Task")).isTrue();
+        } catch (Exception e) {
+            ut.rollback();
+            throw e;
+        }
+        ut.rollback();
+        ut = InitialContext.doLookup(USER_TRANSACTION_NAME);
+        ut.begin();
+
+        try {
```

Comment:
This and the above try-catch block could be likely refactored to a standalone method.